sdk/nodejs/output.ts

```diff
@@ -242,21 +264,26 @@ This function may throw in a future version of @pulumi/pulumi.`;
 To manipulate the value of this Output, use '.apply' instead.`);
     }
 
-    public apply<U>(func: (t: T) => Input<U>): Output<U> {
-        const applied = Promise.all([this.promise(), this.isKnown, this.isSecret])
-                               .then(([value, isKnown, isSecret]) => applyHelperAsync<T, U>(value, isKnown, isSecret, func));
+    // runWithUnknowns requests that `func` is run even if `isKnown` resolves to `false`. This is used to allow
+    // callers to process fully- or partially-unknown values and return a known result. the output proxy takes
+    // advantage of this to allow proxied property accesses to return known values even if other properties of
+    // the containing object are unknown.
+    public apply<U>(func: (t: T) => Input<U>, runWithUnknowns?: boolean): Output<U> {
+        const applied = Promise.all([this.promise(/*withUnknowns*/ true), this.isKnown, this.isSecret])
```

Comment:
so it's unclear to me why you souldn't just pass `runWithUnknowns` in to `this.promise()`.  If the user is not passing a value in (or passing in false), then we wouldn't want unknowns from the underlying promise right?  We'd only care about htem if the user actually passed in `true` here, right?  in other words, unilaterally passing in `true` here doesn't seem necessary.   if it is, can we doc why?